core/src/banking_stage/consume_worker.rs

```diff
@@ -81,7 +81,10 @@ impl<Tx: TransactionWithMeta> ConsumeWorker<Tx> {
             .fetch_add(get_bank_us, Ordering::Relaxed);
 
         for work in try_drain_iter(work, &self.consume_receiver) {
-            if bank.is_complete() {
+            if bank.is_complete() || {
+                // check if the bank got interrupted before completion
```

Comment:
for my own understanding: when can a bank get interrupted before completion?

Reply:
During replay stage we could have discovered a better fork to build our leader blocks on top of. When that happens we immediately abandon the current block we are building and never complete it.

Reply:
I see thank you!